vizro-core/tests/unit/vizro/models/_components/form/test_slider.py

```diff
@@ -0,0 +1,220 @@
+
+
+@pytest.fixture()
+def expected_slider():
+    return dcc.Slider(
+        min=0,
+        max=10,
+        step=1,
+        marks={},
+        value=5,
+        included=False,
+        className="slider_control",
+        id="slider_id",
+        persistence=True,
+    )
+
+
+class TestSliderInstantiation:
+    """Tests model instantiation."""
+
+    def test_create_slider_mandatory(self):
+        slider = vm.Slider()
+
+        assert hasattr(slider, "id")
+        assert slider.type == "slider"
+        assert slider.min is None
+        assert slider.max is None
+        assert slider.step is None
+        assert slider.marks is None
+        assert slider.value is None
+        assert slider.title is None
+        assert slider.actions == []
+
+    @pytest.mark.parametrize(
+        "min, max",
+        [
+            (0, None),
+            (None, 10),
+            (0, 10),
+        ],
+    )
+    def test_valid_min_max(self, min, max):
+        slider = vm.Slider(min=min, max=max)
+
+        assert slider.min == min
+        assert slider.max == max
+
+    def test_invalid_min_max(self):
+        with pytest.raises(
+            ValidationError,
+            match="Maximum value of slider is required to be larger than minimum value.",
+        ):
+            vm.Slider(min=10, max=0)
+
+    @pytest.mark.parametrize(
+        "value",
+        [
+            5,
+            6.5,
+            0,
+            10,
+        ],
+    )
+    def test_valid_value(self, value):
+        slider = vm.Slider(min=0, max=10, value=value)
+
+        assert slider.value == value
+
+    def test_invalid_value_low(self):
+        with pytest.raises(
+            ValidationError,
+            match="Please provide a valid value larger than the minimum value.",
+        ):
+            vm.Slider(min=1, max=10, value=0)
+
+    def test_invalid_value_high(self):
+        with pytest.raises(
+            ValidationError,
+            match="Please provide a valid value smaller than the maximum value.",
+        ):
+            vm.Slider(min=1, max=10, value=11)
+
+    @pytest.mark.parametrize(
+        "step",
+        [
+            1,
+            2.5,
+            10,
+            15,
+        ],
+    )
+    def test_valid_step(self, step):
+        slider = vm.Slider(min=0, max=10, step=step)
+
+        assert slider.step == step
+
+    def test_valid_marks_default(self):
+        slider = vm.Slider(min=0, max=10, step=1)
+
+        assert slider.marks == {}
+
+    @pytest.mark.parametrize(
+        "marks, expected",
+        [
+            (
+                {str(i): i for i in range(0, 10, 5)},
+                {str(i): i for i in range(0, 10, 5)},
+            ),
+            ({15: 15, 25: 25}, {"15": 15.0, "25": 25.0}),
+        ],
+    )
+    def test_valid_marks(self, marks, expected):
+        slider = vm.Slider(min=0, max=10, marks=marks)
+
+        assert slider.marks == expected
+
+    def test_invalid_marks(self):
```

Comment:
What happens when you provide both steps and marks to slider?

Reply:
That is covered in `test_valid_marks_default`, but maybe it is not 100% clear, so I renamed it to `test_valid_marks_with_step`